components/contact-details.tsx

```diff
@@ -35,10 +38,23 @@ const DescriptionItem = (props: DescriptionItemProps) => (
 type DescriptionLinkProps = {
   label: string;
   value?: string;
+  contact: Contact;
+  provinceName: string;
 };
 
-const DescriptionLink = (props: DescriptionLinkProps) =>
-  props.value ? (
+const DescriptionLink = (props: DescriptionLinkProps) => {
+  const { keterangan, penyedia, kontak } = props.contact;
+
+  const formBaseUrl =
+    "https://docs.google.com/forms/d/e/1FAIpQLSeDdZLMD9RTfOiKMItNb8542KAO5w3x9O2ZFQmNyff1rYRZvQ/viewform?usp=pp_url";
+
+  const formQuery = `&entry.346789668=__other_option__&entry.346789668.other_option_response=${keterangan}&entry.323081545=${props.provinceName}&entry.68818336=${penyedia}&entry.217416134=${kontak}`;
+
+  const reportButtonHandler = () => {
+    window.open(formBaseUrl.concat(encodeURI(formQuery)), "_blank");
+  };
```

Comment:
I would suggest you extract the `reportButtonHandler` functionality into a separate function so it is reusable for both `DescriptionLink` and `DescriptionItem` components.